### DISCUSSION TITLE
Is there way to save some image as grayscale PNG? Or as 1-bit PNG?

### DISCUSSION THREAD

**Discussion by Alepod** [ImageMagick/ImageMagick#2819] [Help]
Is there way to save some image as grayscale PNG? Or as 1-bit PNG? Like this PNG for example
![874_-_Mad_Skull_car_decal__12297_zoom](https://user-images.githubusercontent.com/22130090/98986544-1a9eab80-2536-11eb-8161-87d82448db65.png)

**Comment by fmw42** [comment]
One way on IM 6 is

`convert image -threshold 50% result.png`

Or if already as desired as b/w, then

`convert image -type bilevel result.png`

or if b/w but you want it grayscale, then

`convert image -type grayscale result.png`

For IM 7, use magick rather than convert.

What is the issue?  What is your input image? What is your version of IM?  What platform?

**Comment by t246246** [comment]
Your attached image has bit-depth=1 color-type=3. https://www.w3.org/TR/PNG/#11IHDR
If you want exactly png that has these types, you can explicitly use -define png:color-type=3 and -define png:bit-depth=1.

One minor notice.  Sometimes when you start from some color image, you may see this message.

$ magick logo: -threshold 50% -type bilevel -define png:color-type=3 -define png:bit-depth=1 logo.png
magick.exe: Cannot write image with defined png:bit-depth or png:color-type. `logo.png' @ warning/png.c/MagickPNGWarningHandler/1748.

As per spec, this is allowed combination.  After little trial this seems working.

$ magick logo: -threshold 50% -type bilevel -background none -define png:color-type=3 -define png:bit-depth=1 logo.png
